ui/app/AppLayouts/Profile/controls/WalletKeyPairDelegate.qml

```diff
@@ -0,0 +1,93 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.14
+
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+import utils 1.0
+
+Rectangle {
+    id: root
+
+    property string chainShortNames
+    property string userProfilePublicKey
+
+    signal goToAccountView(var account)
+
+    QtObject {
+        id: d
+        property var relatedAccounts: model.keyPair.accounts
+        property bool isWatchOnly: model.keyPair.pairType === Constants.keycard.keyPairType.watchOnly
+        property bool isPrivateKeyImport: model.keyPair.pairType === Constants.keycard.keyPairType.privateKeyImport
+        property bool isProfileKeypair: model.keyPair.pairType === Constants.keycard.keyPairType.profile
+        property string locationInfo: model.keyPair.migratedToKeycard ? qsTr("On Keycard"): qsTr("On device")
+    }
+
+    implicitHeight: layout.height
+    color: Theme.palette.baseColor4
+    radius: 8
+
+    ColumnLayout {
+        id: layout
+        anchors.horizontalCenter: parent.horizontalCenter
+        width: parent.width
+        StatusListItem {
+            Layout.fillWidth: true
+            title: d.isWatchOnly ? qsTr("Watch only") : model.keyPair.name
+            statusListItemSubTitle.textFormat: Qt.RichText
+            titleTextIcon: model.keyPair.migratedToKeycard ? "keycard": ""
+            subTitle: d.isWatchOnly ? "" : d.isProfileKeypair ?
+                      Utils.getElidedCompressedPk(model.keyPair.pubKey) + Constants.settingsSection.dotSepString + d.locationInfo : d.locationInfo
+            color: Theme.palette.transparent
+            ringSettings {
+                ringSpecModel: d.isProfileKeypair ? Utils.getColorHashAsJson(root.userProfilePublicKey) : []
+                ringPxSize: Math.max(asset.width / 24.0)
+            }
+            asset {
+                width: model.keyPair.icon ? 24 : 40
+                height: model.keyPair.icon ? 24 : 40
+                name: model.keyPair.image ? model.keyPair.image : model.keyPair.icon
+                isImage: !!model.keyPair.image
+                color: d.isProfileKeypair ? Utils.colorForPubkey(root.userProfilePublicKey) : Theme.palette.primaryColor1
+                letterSize: Math.max(4, asset.width / 2.4)
+                charactersLen: 2
+                isLetterIdenticon: !model.keyPair.icon && !asset.name.toString()
+            }
+            components: [
+                StatusFlatRoundButton {
+                    icon.name: "more"
+                    icon.color: Theme.palette.directColor1
+                    visible: !d.isWatchOnly
+                },
+                StatusBaseText {
```

Comment:
`StatusSwitch` can take a text, move it there directly as `text: foo`

Reply:
the text assigns itself to the right not left of the checkbox 

![image](https://github.com/status-im/status-desktop/assets/60327365/28acd6f5-4d0f-4dae-b87e-73e81187ff16)

Reply:
@caybro Ive confirmed with Ben and the text should be to the left of the checkbox. Please approve when you can :)